### PR TITLE
Add --audio-only flag

### PR DESCRIPTION
**Pull request by GNUSheep**
I think this option is more user friendly than choosing audio stream from streams output.

I tested it out with youtube videos, and it's working pretty fine.

### CODE REVIEW

File: downloader/downloader.go [hunk 1]
```diff
@@ -568,6 +569,20 @@ func (downloader *Downloader) Download(data *extractors.Data) error {
 		return errors.Errorf("no stream named %s", streamName)
 	}
 
+	if downloader.option.AudioOnly {
+		for _, s := range sortedStreams {
+			// Looking for the best quality
+			matches, ok := regexp.MatchString("audio", s.Quality)
+			if ok != nil {
+				return errors.Errorf("No audio stream found")
```
Comment (iawia002) [inline downloader/downloader.go:577]: This error message is so misleading, I think we should just return the error returned by the `MatchString` function.

```suggestion
			if err != nil {
				return err
```

File: downloader/downloader.go [hunk 2]
```diff
@@ -568,6 +569,20 @@ func (downloader *Downloader) Download(data *extractors.Data) error {
...
+			}
+			if matches {
+				stream, _ = data.Streams[s.ID]
+				break
+			}
+		}
```
Comment (iawia002) [inline downloader/downloader.go:583]: And we need to end the function here, if the audio stream is not found

**Comment by GNUSheep** [comment]
Thanks for the review, I resolved all proposed changes.

File: downloader/downloader.go [hunk 3]
```diff
@@ -568,6 +569,25 @@ func (downloader *Downloader) Download(data *extractors.Data) error {
 		return errors.Errorf("no stream named %s", streamName)
 	}
 
+	if downloader.option.AudioOnly {
+		var isFound bool
+		for _, s := range sortedStreams {
+			// Looking for the best quality
+			matches, err := regexp.MatchString("audio", s.Quality)
```
Comment (iawia002) [inline downloader/downloader.go:576]: Lint failed here, can you take a look?

**Comment by GNUSheep** [comment]
Thanks again for review. I fixed errors, now everything should be all right

**Review by iawia002 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**